tests/jlm/hls/backend/rvsdg2rhls/MemoryConverterTests.cpp

```diff
@@ -0,0 +1,385 @@
+      jlm::util::AssertedCast<jlm::rvsdg::node_output>(lambdaRegion->result(1)->origin())->node();
+  jlm::util::AssertedCast<const LambdaExitMemoryStateMergeOperation>(
+      &exitMemoryStateMergeNode->operation());
+
+  // Act
+  ConvertThetaNodes(*rvsdgModule);
+  // Simple assert as ConvertThetaNodes() is tested in separate unit tests
+  jlm::rvsdg::view(rvsdgModule->Rvsdg(), stdout);
+  assert(jlm::rvsdg::Region::Contains<loop_op>(*lambdaRegion, true));
+
+  // Act
+  mem_queue(*rvsdgModule);
+  // Simple assert as mem_queue() is tested in separate unit tests
+  jlm::rvsdg::view(rvsdgModule->Rvsdg(), stdout);
+  assert(jlm::rvsdg::Region::Contains<state_gate_op>(*lambdaRegion, true));
+  assert(jlm::rvsdg::Region::Contains<MemoryStateSplitOperation>(*lambdaRegion, true));
+  assert(jlm::rvsdg::Region::Contains<MemoryStateMergeOperation>(*lambdaRegion, true));
+
+  // Act
+  MemoryConverter(*rvsdgModule);
+  // Assert
+  jlm::rvsdg::view(rvsdgModule->Rvsdg(), stdout);
+
+  // Memory Converter replaces the lambda so we start from the root of the graph
+  auto region = rvsdgModule->Rvsdg().root();
+  assert(region->nnodes() == 1);
+  lambda = jlm::util::AssertedCast<jlm::llvm::lambda::node>(region->nodes.first());
+  lambdaRegion = lambda->subregion();
+
+  assert(jlm::rvsdg::Region::Contains<mem_resp_op>(*lambdaRegion, true));
+  assert(jlm::rvsdg::Region::Contains<mem_req_op>(*lambdaRegion, true));
+
+  // Request Node
+  auto requestNode =
+      jlm::util::AssertedCast<jlm::rvsdg::node_output>(lambdaRegion->result(2)->origin())->node();
+  jlm::util::AssertedCast<const mem_req_op>(&requestNode->operation());
+
+  // HLS_LOOP Node
+  auto loopOutput =
+      jlm::util::AssertedCast<const jlm::rvsdg::structural_output>(requestNode->input(0)->origin());
+  auto loopNode = jlm::util::AssertedCast<const jlm::rvsdg::StructuralNode>(loopOutput->node());
+  jlm::util::AssertedCast<const loop_op>(&loopNode->operation());
+  // Loop Result
+  auto & thetaResult = loopOutput->results;
+  assert(thetaResult.size() == 1);
+  // Load Node
+  auto loadNode =
+      jlm::util::AssertedCast<const jlm::rvsdg::node_output>(thetaResult.first()->origin())->node();
+  jlm::util::AssertedCast<const decoupled_load_op>(&loadNode->operation());
+  // Loop Argument
+  auto thetaArgument =
+      jlm::util::AssertedCast<const jlm::rvsdg::RegionArgument>(loadNode->input(1)->origin());
+  auto thetaInput = thetaArgument->input();
+
+  // Response Node
+  auto responseNode =
+      jlm::util::AssertedCast<const jlm::rvsdg::node_output>(thetaInput->origin())->node();
+  jlm::util::AssertedCast<const mem_resp_op>(&responseNode->operation());
+
+  // Lambda argument
+  jlm::util::AssertedCast<const jlm::rvsdg::RegionArgument>(responseNode->input(0)->origin());
+}
+
+static int
+Tests()
+{
+  TestTraceArgument();
+  TestLoad();
+  TestLoadStore();
+  TestThetaLoad();
+
+  return 0;
+}
+
+JLM_UNIT_TEST_REGISTER("jlm/hls/backend/rvsdg2rhls/MemoryConverterTests", Tests)
```

Comment:
Could you register each test individually with `JLM_UNIT_TEST_REGISTER`, please. You can remove the `Tests()` method and this registration then afterwards.

Reply:
Maybe move the registration to the respective function so they are together and one does not need to search for it.

Reply:
Moved